### ISSUE TITLE
Crash while duplicating layer

### ISSUE THREAD

**Issue by qkdreyer** [deepnight/led#193]
Crash while duplicating layer, removing previous layer, and renaming duplicated layer to previous layer

Stack:
```
LEd version: 0.5.0-beta
Missing layer instance for 21
Error

```

Log:
```
[WARNING]      Full render requested
[EVENT]        LevelResized
[WARNING]      All rules applied in Instance[#47,AutoLayer:AutoLayer]
[WARNING]      All rules applied in Instance[#21,Collisions:IntGrid]
[WARNING]      All rules applied in Instance[#89,Background:IntGrid]
[WARNING]      Full render requested
[EVENT]        LevelResized
[WARNING]      All rules applied in Instance[#47,AutoLayer:AutoLayer]
[WARNING]      All rules applied in Instance[#21,Collisions:IntGrid]
[WARNING]      All rules applied in Instance[#89,Background:IntGrid]
[WARNING]      Full render requested
[EVENT]        LevelResized
[WARNING]      All rules applied in Instance[#47,AutoLayer:AutoLayer]
[WARNING]      All rules applied in Instance[#21,Collisions:IntGrid]
[WARNING]      All rules applied in Instance[#89,Background:IntGrid]
[WARNING]      Full render requested
[FILE]         Reading HTML template editLayerDefs from cache
[EVENT]        LayerInstanceSelected
[RENDER]       Rendered bg
[EVENT]        LayerInstanceSelected
[RENDER]       Rendered bg
[TIDY]         Added missing layer instance Collisions2 in data.Level.FirstLevel(#22)
[TIDY]         Added missing layer instance Collisions2 in data.Level.DNFirstLevel(#0)
[GENERAL]      "Updating auto-layers...", 2 operation(s):
[GENERAL]        - FirstLevel.Collisions2 (50%)
[EVENT]        LayerDefAdded
[WARNING]      All rules applied in Instance[#101,Collisions2:IntGrid]
[GENERAL]        - DNFirstLevel.Collisions2 (100%)
[WARNING]      All rules applied in Instance[#101,Collisions2:IntGrid]
[EVENT]        LayerInstanceSelected
[RENDER]       Rendered bg
[EVENT]        LayerInstanceSelected
[RENDER]       Rendered bg
[EVENT]        LayerInstanceSelected
[RENDER]       Rendered bg
[EVENT]        LayerInstanceSelected
[RENDER]       Rendered bg
[WARNING]      Last chance for: Layer Collisions deleted
[TIDY]         Removed lost layer instance in data.Level.FirstLevel(#22)
[TIDY]         Removed lost layer instance in data.Level.DNFirstLevel(#0)
[EVENT]        LayerInstanceSelected
[EVENT]        LayerDefRemoved
[RENDER]       Rendered bg
[EVENT]        LayerInstanceSelected
[RENDER]       Rendered bg
[EVENT]        LayerDefChanged
[FILE]         Loading HTML template crash
[ERROR]        Missing layer instance for 21 (Error)
[ERROR]        undefined)

```

**Comment by deepnight** [comment]
I see the problem 🤔  It's because you have an **Auto layer** that uses the **deleted IntGrid layer** as source for its int values. 

I'll support this kind of operation in 0.5.0 . I'll try to push the update asap :)

**Comment by deepnight** [comment]
Fixed in 0.5.0 (not released yet)
